app/models/live_metric.rb

```diff
@@ -8,18 +8,20 @@ def self.find(*args)
     validate_raw_query(raw_query)
     processed = process_conditions(raw_query[:conditions])
     resource = fetch_resource(processed[:resource_type], processed[:resource_id])
+    ext_options = raw_query[:ext_options]
+    filtered_cols = ext_options ? ext_options[:only_cols] : nil
```

Comment:
Minor, but if you don't need the ext_options (i.e. it's a temporary variable), then you can simplify these lines to

```ruby
filtered_cols = raw_query.fetch_path(:ext_options, :only_cols)
```